devmapper: Drop redundant lock and redundant function

rhvgoyal opened the pull request:
This PR does two things.
- Drops unused function HasActivatedDevice()
- Gets rid of a redundant lock and replaces it with global devices lock. This makes code easier to read and makes locking semantics clearer.

rhvgoyal committed:
devmapper: Get rid of unused function HasActivatedDevice()

Looks like nobody is calling HasActivatedDevice(). Get rid of it.

Signed-off-by: Vivek Goyal <vgoyal@redhat.com>

rhvgoyal commented:
CC @LK4D4 @calavera @vbatts

vbatts commented:
LGTM

cpuguy83 commented on daemon/graphdriver/devmapper/deviceset.go:
```diff
@@ -370,6 +368,13 @@ func (devices *DeviceSet) lookupDevice(hash string) (*devInfo, error) {
 	return info, nil
 }
 
+func (devices *DeviceSet) lookupDeviceLock(hash string) (*devInfo, error) {
```
how about `lookupDeviceWithLock`

rhvgoyal replied:
sounds good. Will change.

rhvgoyal committed:
devmapper: Get rid of metaData.devicesLock

Right now we seem to have 3 locks. 

- devinfo.lock
  This is a per device lock

- metaData.devicesLock

  This is supposedely protecting map of devices.

- Global DeviceSet lock

  This is protecting map of devices as well as serializing calls to libdevmapper.

Semantics of per devices lock and global deviceset lock seem to be very clear.
Even ordering between these two locks has been defined properly.

What is not clear is the need and ordering of metaData.devicesLock. Looks like
this lock is not necessary and global DeviceSet lock should be used to
protect map of devices as it is part of DeviceSet.

This patchset gets rid of metaData.devicesLock and instead uses DeviceSet
lock to protect map of devices.

Also at couple of places during initialization takes devices.Lock(). That
is not strictly necessary as there is supposed to be one thread of execution
during initializaiton. Still it makes the code clearer.

I think this makes code more clear and easier to understand and easier to
make further changes.

Signed-off-by: Vivek Goyal <vgoyal@redhat.com>

rhvgoyal commented:
Renamed to lookupDeviceWithLock() and pushed the patch again.

LK4D4 commented:
LGTM
